api/types.proto

```diff
@@ -299,20 +316,27 @@ message UpdateStatus {
 		UPDATING = 1;
 		PAUSED = 2;
 		COMPLETED = 3;
-		// TODO(aaronl): add ROLLING_BACK, ROLLED_BACK as part of
-		// rollback support.
+		ROLLING_BACK = 4;
+		ROLLBACK_PAUSED = 5; // if a rollback fails
+		ROLLBACK_COMPLETED = 6;
 	}
 
 	// State is the state of this update. It indicates whether the
-	// update is in progress, completed, or is paused.
+	// update is in progress, completed, paused, rolling back, or
+	// finished rolling back.
 	UpdateState state = 1;
 
 	// StartedAt is the time at which the update was started.
 	Timestamp started_at = 2;
 
-	// CompletedAt is the time at which the update completed.
+	// CompletedAt is the time at which the update completed successfully,
+	// paused, or finished rolling back.
 	Timestamp completed_at = 3;
 
+	// StartedRollingBackAt is the time at which the update started rolling
+	// back to the previous version.
+	Timestamp started_rolling_back_at = 5;
+
```

Comment:
From a user's perspective, I don't think it's fair to reject a service update because the number of satisfied slots changed while the "docker service update" command was running - especially because the user has no non-racy way of doing a check-and-set based on this information. I think we should either put in automatic retries or a spec version to avoid this user-facing issue.